openbsd/Platform.c

```diff
@@ -216,45 +211,10 @@ void Platform_setMemoryValues(Meter* this) {
    this->values[2] = cachedMem;
 }
 
-/*
- * Copyright (c) 1994 Thorsten Lockert <tholo@sigmasoft.com>
- * All rights reserved.
- *
- * Taken almost directly from OpenBSD's top(1)
- */
```

Comment:
How's the copyright situation of this? `All rights reserved.` usually doesn't imply and license grant to use the code?

Reply:
That seems to be a BSD-license import from https://github.com/openbsd/src/blob/master/usr.bin/top/machine.c#L818

I suggest to keep this header but amend:

```
* Originally released under a BSD-3 license
* Modified through htop developers applying GPL-2
```

The code has been significantly modified through htop developers, so it is licensed GPL-2 now. But it does (in my non-professional assessment) not constitute a separate work as the original structure and logic is retained. So we should keep the original copyright claim.
Read: Somebody would likely be free to copy these pieces of code again and claim their use as fair under BSD-3.

Reply:
Okay, than this updated copyright notice should go to the new location of this piece of code, as this is just the location where the code was removed from.

Reply:
The *new* code is in OpenBSDProcessList.c:
```c
   int nswap = swapctl(SWAP_NSWAP, 0, 0);
   if (nswap > 0) {
      struct swapent swdev[nswap];
      int rnswap = swapctl(SWAP_STATS, swdev, nswap);

      /* Total things up */
      unsigned long long int total = 0, used = 0;
      for (int i = 0; i < rnswap; i++) {
         if (swdev[i].se_flags & SWF_ENABLE) {
            used += (swdev[i].se_inuse / (1024 / DEV_BSIZE));
            total += (swdev[i].se_nblks / (1024 / DEV_BSIZE));
         }
      }

      pl->totalSwap = total;
      pl->usedSwap = used;
   } else {
      pl->totalSwap = pl->usedSwap = 0;
   }
```

In my opinion (IANAL) one can not claim copyright on trivial code or on code than can be almost only written in one way.
How would one else write code for OpenBSD to get swap statistics?  see https://man.openbsd.org/man2/swapctl.2

Reply:
What's so bad about just adding in the two lines suggested by @fasterit to give attribution of where that source originally came from?

I know of a case (in the US) where someone got some DMCA over 5 assembly instructions for swapping two registers, because they coincided with the source code of the original manufacturer and being placed the exact same location as programs where expecting this function at that location …

So: Doesn't hurt to include some attribution here, even if the original code has been heavily modified by us already.

Reply:
Added copyright note